### ISSUE TITLE
About computing metrics

### ISSUE THREAD

**Issue by mkstmyk** [athn-nik/motionfix#9]
Thank you for the excellent code and data!
When I evaluated the generated results of the distributed pre-trained model using the following command:
python

> compute_metrics.py folder=experiments/tmed/3way_steps_300_motionfix_noise_last/ld_txt-2.0_ld_mot-2.0

I obtained the following results:

> ===== Metrics for Retrieval on the full test set =====
> {'R@1_s2t': '59.46', 'R@2_s2t': '75.68', 'R@3_s2t': '82.43', 'R@1': '47.30', 'R@2': '70.27', 'R@3': '77.03'}

I think these numbers are not as good as those reported in the paper. Is there any parameter adjustment needed to obtain the same results as in the paper?

**Comment by atnikos** [comment]
Hello, I guess this is the full test which I am computing in the supmat (the main papers report the batches-32 setting).
Lmk,  if you have any issues and I should re-check my evaluation. The normal results (batches) should be around 60% r@1 gen2target.

**Comment by mkstmyk** [comment]
Thank you! I re-read the paper and understood it. I apologize for missing the batch description in the paper!
I obtained results close to the values in the paper as follows:

> ===== Metrics for Retrieval on Batches of 32 =====
> {'R@1_s2t': '76.56', 'R@2_s2t': '90.62', 'R@3_s2t': '93.75', 'R@5_s2t': '98.44', 'R@10_s2t': '100.0', 'MedR_s2t': '1.00', 'AvgR_s2t': '1.50', 'R@1': '56.25', 'R@2': '81.25', 'R@3': '84.38', 'R@5': '90.62', 'R@10': '92.19', 'MedR': '1.00', 'AvgR': '2.64'}
> 
Also, why does the paper present the search results for a batch size of 32, even though the proposed method achieves good numbers on the full data search?

I have one more question: Is R@1_s2t the value for gen2target, and R@1 the value for gen2source? When I look at motion2motion_retr.py, it seems like the values are for source2target and target2gen based on the following description, but is the actual calculation the same as in the paper, i.e., gen2target and gen2source?
```
mat2name = {
            'sim_matrix_s_t': 'source_target',
            'sim_matrix_t_t': 'target_generated'
            }
```

**Comment by atnikos** [comment]
Batches of random 32 elements are more reasonable in the sense that you get less ambiguous retrievals. To give an example:
When you have motion A in test set and you want to see what its features retrieve, and this motion exists in the target motions you might retrieve yourself, or a crop that is close to yourself (a very similar motion). That is why testing it on batches is more fair as you retrieve in a smaller set, which is much harder to have very similar motions. The problem is that when evaluating we are not using the text ( that is why we are using gen2source score). So you want in the end to retrieve the corresponding target while you are close enough to the source ( because it is editing and not generation). Ideally you should have the text involved in the evaluation and maybe a positional score, which we report briefly in supmat.

**Comment by atnikos** [comment]
Yes, the evaluation is same as in the the paper reg. the metrics. Because your retrieval gallery is the test set. Naming in the code is a bit unfortunate. I will check it and fix it.

**Comment by mkstmyk** [comment]
Thank you for the detailed explanation. I am finally starting to understand.
Is my understanding correct that batch processing acts as a kind of filtering because, with a search on the full dataset, there is a higher chance of "coincidentally" obtaining motions very close to the query motion? 

Also, I have a question about the distributed checkpoint. The evaluation values were obtained as follows:

> ===== Metrics for Retrieval on Batches of 32 =====
{'R@1_s2t': '76.56', 'R@2_s2t': '90.62', 'R@3_s2t': '93.75', 'R@5_s2t': '98.44', 'R@10_s2t': '100.0', 'MedR_s2t': '1.00', 'AvgR_s2t': '1.50', 'R@1': '56.25', 'R@2': '81.25', 'R@3': '84.38', 'R@5': '90.62', 'R@10': '92.19', 'MedR': '1.00', 'AvgR': '2.64'}
===== Metrics for Retrieval on the full test set =====
{'R@1_s2t': '59.46', 'R@2_s2t': '75.68', 'R@3_s2t': '82.43', 'R@5_s2t': '90.54', 'R@10_s2t': '97.30', 'MedR_s2t': '1.00', 'AvgR_s2t': '2.46', 'R@1': '47.30', 'R@2': '70.27', 'R@3': '77.03', 'R@5': '85.14', 'R@10': '90.54', 'MedR': '2.00', 'AvgR': '4.24'} 

Comparing these with the evaluation values of the model trained on the same motionfix.pth.tar without any changes to the code or settings:

> ===== Metrics for Retrieval on Batches of 32 =====
{'R@1_s2t': '73.54', 'R@2_s2t': '86.25', 'R@3_s2t': '88.75', 'R@5_s2t': '94.17', 'R@10_s2t': '98.33', 'MedR_s2t': '1.00', 'AvgR_s2t': '1.91', 'R@1': '59.79', 'R@2': '74.17', 'R@3': '80.42', 'R@5': '86.67', 'R@10': '93.75', 'MedR': '1.20', 'AvgR': '3.02'}
===== Metrics for Retrieval on the full test set =====
{'R@1_s2t': '32.02', 'R@2_s2t': '44.66', 'R@3_s2t': '54.94', 'R@5_s2t': '64.03', 'R@10_s2t': '77.08', 'MedR_s2t': '3.00', 'AvgR_s2t': '14.77', 'R@1': '17.79', 'R@2': '28.85', 'R@3': '37.94', 'R@5': '49.01', 'R@10': '60.47', 'MedR': '6.00', 'AvgR': '33.76'}

While the performance on the batch approaches the distributed checkpoint, the evaluation on the full dataset is significantly lower.
What could be the reason for this? Is there any need for adjustments in the training settings? I would appreciate it if you could tell me the optimal batch size and learning rate, if any.

Regarding the evaluation values, is my understanding correct that 'sim_matrix_s_t': 'source_target' actually corresponds to gen2source, and 'sim_matrix_t_t': 'target_generated' corresponds to gen2target?
I apologize for asking so many questions, but thank you in advance for your assistance.

**Comment by atnikos** [comment]
I am always happy to answer questions :) I will check my training parameters and get back to you soon. 
The batch metrics are more important, both gen2source and gen2target. I report the metrics on the full dataset for completeness, but I believe batches are enough to check on performance and compare. 
I think metric that takes into account the source motion and the edit text would be ideal, but I am looking into it for the future.

--

Yes, your understanding of the matrices is correct.
